delta-api/src/main/java/io/cdap/delta/api/assessment/TableNotFoundException.java

```diff
@@ -25,7 +25,19 @@ public TableNotFoundException(String database, String table, String errorMessage
     super(String.format("Table '%s' in database '%s' was not found. %s", table, database, errorMessage));
```

Comment:
This can just call `this(database, null, table, errorMessage)` to avoid duplicate message creation.

Reply:
Actually I did that on purpose , because the schema appears in the middle of the error message
to construct the error message based on whether the schema is null will make the code clumsy and not that readable

Reply:
for example, if we do below
```
if (schema != null) 
  errorMessage = String.format("Table '%s' in database '%s' and schema '%s' was not found. %s", table, database, schema, errorMessage)
else 
 errorMessage = String.format("Table '%s' in database '%s' was not found. %s", table, database, errorMessage)
```
Then it has no difference from my current implementation

Reply:
If we want to share the comment part , then assembling messages will make code clumsy and less readable :
```
message = String.format("Table '%s' in database '%s' ", table, database);
if (schema != null) {
   message = String.format ("%s and schema '%s'" , message, schema);
}
message = String.format("%s  was not found. %s", message, errorMessage);
```
What do you think ?

Reply:
It is not a big deal here but in general since `TableNotFoundException` is a part of API, it is easy for client of it to call the version (database, schema, table) with schema as `null` by mistake. API does not restrict client from doing so. To make it strict, we can check if schema is  `null` and throw an exception but that would be overkill for this case. Another way to make it stricter is implementation can handle `null` correctly. 

We can do something like following right?
```
String schemaInfo = schema == null ? "" : String.format(" and schema %s ", schema);
String.format("Table '%s' in database '%s'%s was not found.", table, database, schemaInfo)
```

Also this avoids un-necessary code duplication.

Reply:
sg, done :)